src/plugins/Transloadit/index.js

```diff
@@ -281,6 +304,191 @@ module.exports = class Transloadit extends Plugin {
     })
   }
 
+  getPersistentData (setData) {
+    const state = this.getPluginState()
+    const assemblies = state.assemblies
+    const uploadsAssemblies = state.uploadsAssemblies
+    const uploads = Object.keys(state.files)
+    const results = state.results.map((result) => result.id)
+
+    setData({
+      [this.id]: {
+        assemblies,
+        uploadsAssemblies,
+        uploads,
+        results
+      }
+    })
+  }
+
+  /**
+   * Emit the necessary events that must have occured to get from the `prevState`,
+   * to the current state.
+   * For completed uploads, `transloadit:upload` is emitted.
+   * For new results, `transloadit:result` is emitted.
+   * For completed or errored assemblies, `transloadit:complete` or `transloadit:assembly-error` is emitted.
+   */
+  emitEventsDiff (prevState) {
+    const opts = this.opts
+    const state = this.getPluginState()
+
+    const emitMissedEvents = () => {
+      // Emit events for completed uploads and completed results
+      // that we've missed while we were away.
+      const newUploads = Object.keys(state.files).filter((fileID) => {
+        return !prevState.files.hasOwnProperty(fileID)
+      }).map((fileID) => state.files[fileID])
+      const newResults = state.results.filter((result) => {
+        return !prevState.results.some((prev) => prev.id === result.id)
+      })
+
+      this.uppy.log('[Transloadit] New fully uploaded files since restore:', newUploads)
+      newUploads.forEach(({ assembly, uploadedFile }) => {
```

Comment:
The way `log()` currently works, you need to call it with the object separately for it to show it, like:

```js
this.uppy.log('[Transloadit] New fully uploaded files since restore:')
this.uppy.log(newUploads)
```

Not ideal, I know :)

Reply:
oops, i just `s/console.log/this.uppy.log`d that. thanks!